### PR TITLE
Use eval instead of $.script

### PR DESCRIPTION
**Pull request by LukasReschke**
Fixes https://github.com/nextcloud/server/issues/4067

Signed-off-by: Lukas Reschke <lukas@statuscode.ch>

@michaelletzgus Mind giving this a try?

### CODE REVIEW

**Comment by michaelletzgus** [comment]
Works :-)

But why `async:false`? At a first glance `async:true` seems to work either...

**Comment by LukasReschke** [comment]
> But why `async:false`? At a first glance `async:true` seems to work either...

I want to be certain that code called after this certainly has the JS dependencies loaded. Not really a  JS expert myself so it could be that the `async:false` is not required. But unless we're 100% sure we should rather stay on the safe side here IMO.

**Comment by michaelletzgus** [comment]
`$.getScript()` is also async by default and it worked before.
The only problem was the dysfunc eval()

**Comment by LukasReschke** [comment]
@michaelletzgus Fair enough – changed in https://github.com/nextcloud/server/pull/4068/commits/fea21087104280cf3245d7916155c87e52de44ca

File: core/js/js.js [hunk 1]
```diff
@@ -366,15 +366,28 @@ var OCP = {},
 	 */
 	addScript:function(app,script,ready){
 		var deferred, path=OC.filePath(app,'js',script+'.js');
-		if(!OC.addScript.loaded[path]){
-			if(ready){
-				deferred=$.getScript(path,ready);
-			}else{
-				deferred=$.getScript(path);
+		if(!OC.addScript.loaded[path]) {
+			if(ready) {
```
Comment (icewind1991) [inline core/js/js.js]: moving the if statement in the callback would clean up the code a bit

**Comment by LukasReschke** [comment]
@icewind1991 Fixed :)

**Review by icewind1991 — APPROVED** [review]

**Comment by michaelletzgus** [comment]
By the way, are there any guidelines in nc coding regarding closures vs. bind? This is a good example because something like this would also work:

```
success: function (content) {
			eval(content);
			if(this.ready) {
				this.ready();
			}.bind(this)
```

**Comment by MorrisJobke** [comment]
> By the way, are there any guidelines in nc coding regarding closures vs. bind? This is a good example because something like this would also work:

No. I would keep it as it is 👍

**Review by MorrisJobke — APPROVED** [review]

**Pull request merged**